### DISCUSSION TITLE
Controlling 18 servos with RPI pico board

### DISCUSSION THREAD

**Discussion by Papouc** [earlephilhower/arduino-pico#1432] [Q&A] [answered]
Hello everyone, 

I would like to ask about arduino-pico Servo library. I need to independently control 18 SG90 servo motors, but it is pretty clear, that it cannot be done using the included library (documentation states 8 servos only). From what I've seen, the servo library used by arduino boards does something like this by borrowing one of the arduino timers and using it to "simulate" PWM signal on any digital pin. Is there any way of achieving something like this with arduino-pico?

Thanks a lot :D!!!

**Comment by earlephilhower** [comment, marked as answer]
Haven't tried it, but there's a library that uses the PWM control to generate servo signals.  Gonna have less resolution than the PIO-based ones, but you should be able to run 16 there (only 8 PWM slices with 2 controllable pins per slice).  Should be able to use the included Servo lib to round out the other two, giving you 18 pins.

See the last comment in https://github.com/earlephilhower/arduino-pico/issues/1396

**Reply by Papouc** [reply]
Thanks for super fast response :D! I'll give it a try.
